test/CadlRanchProjects.Tests/OAuth2TestHelper.cs

```diff
@@ -50,13 +55,55 @@ public override void Process(HttpMessage message, ReadOnlyMemory<HttpPipelinePol
             protected new async ValueTask ProcessNextAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
             {
                 await _transport.ProcessAsync(message).ConfigureAwait(false);
+                await ResponseBodyPolicyAsync(message, pipeline).ConfigureAwait(false);
 
                 var response = message.Response;
                 Type responseType = response.GetType();
                 PropertyInfo propInfo = responseType.GetProperty("IsError", BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic);
                 propInfo.SetValue(response, message.ResponseClassifier.IsErrorResponse(message));
             }
 
+            private async ValueTask ResponseBodyPolicyAsync(HttpMessage message, ReadOnlyMemory<HttpPipelinePolicy> pipeline)
```

Comment:
I think this method name does not explain what this is doing, could we specify?

Reply:
Looks like this method is trying to copy out the content stream of this response to a new memory stream, and then set it back as the content stream of the response.

I think we need some more comment on the method or where we are doing this to explain why we need this.
I assume we must do this because we want the stream could always be seekable. Any way, please add some comment to explain what we are doing here.

Also does the same issue could happen in real world instead of mock tests? Maybe we need to fix our implementation to ensure that as well?

Reply:
Also since the pipeline argument is not used here, please remove it

Reply:
Prepend "Process" to it. I'd like to keep the pipeline parameter to keep consistence with its source.